napari/layers/image/_slice.py

```diff
@@ -12,6 +12,9 @@
 from napari.utils.misc import reorder_after_dim_reduction
 from napari.utils.transforms import Affine
 
+if TYPE_CHECKING:
+    from tensorstore import dtype as ts_dtype
```

Comment:
Does this make tensorstore a requirement for type checking? I'd love to be able to make such dependencies optional even when type checking. (my suggestion below of normalize_dtype would allow torch dtypes etc.)

Reply:
fixed